cuqi/sampler/_laplace_approximation.py

```diff
@@ -130,9 +130,15 @@ def Lk_fun(x_k):
         self._m = len(self._data)
         self._L1 = self.target.likelihood.distribution.sqrtprec
 
+        # If prior location is scalar, repeat it to match dimensions
+        if len(self.target.prior.location) == 1:
+            self._priorloc = np.repeat(self.target.prior.location, self.dim)
+        else:
+            self._priorloc = self.target.prior.location
```

Comment:
Just a suggestion to make `self._priorloc` not class variable but only defined locally in the method e.g.  `_priorloc` instead of  `self._priorloc`. Because it is not used in other methods in this calss, and just in case we provide flexibility to update the prior then this can prevent inconsistency.

Reply:
Most other parameters were stored like that. I agree perhaps all should be updated to be defined in the sample scope, but for now I suggest we keep it like this.